### PR TITLE
Updated Task Initiation DMN to accept appealType as Json

### PR DESCRIPTION
**Pull request by indvamshi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-615

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/resources/wa-task-initiation-ia-asylum.dmn [hunk 1]
```diff
@@ -14,7 +14,10 @@
       </input>
       <input id="InputClause_0a0i7vo" label="Appeal Type" camunda:inputVariable="appealType">
         <inputExpression id="LiteralExpression_18qwedg" typeRef="string">
-          <text></text>
+          <text>if(caseData != null and caseData.Data != null and caseData.Data.appealType != null) then
+caseData.Data.appealType
```
Comment (alexbaeza) [inline src/main/resources/wa-task-initiation-ia-asylum.dmn:18]: is this correct? 
should it be 
```
caseData.data.appealType
```

File: src/test/java/uk/gov/hmcts/reform/iataskconfiguration/dmn/CamundaTaskInitiationTest.java [hunk 2]
```diff
@@ -491,4 +495,20 @@ void if_this_test_fails_needs_updating_with_your_changes() {
         assertThat(logic.getRules().size(), is(24));
 
     }
+
+    private static Map<String, Object> mapAppealType(String appealType) {
+        String appealTypeJson = "{\n"
+               + "   \"Data\":{\n"
```
Comment (alexbaeza) [inline src/test/java/uk/gov/hmcts/reform/iataskconfiguration/dmn/CamundaTaskInitiationTest.java:501]: should this kye be lowercased Data?

File: src/main/resources/wa-task-initiation-ia-asylum.dmn (hunk 1 above)
Comment (indvamshi) [reply src/main/resources/wa-task-initiation-ia-asylum.dmn:18]: In AdditionalData , the data field used as Data json property.
`@ToString
@EqualsAndHashCode
@Builder
public class AdditionalData {

    private final Map<String, Object> data;
    private final Map<String, JsonNode> definition;

    @JsonCreator
    public AdditionalData(@JsonProperty("Data") Map<String, Object> data,
                          @JsonProperty("Definition") Map<String, JsonNode> definition) {
        this.data = data;
        this.definition = definition;
    }

    public Map<String, Object> getData() {
        return data;
    }

    public Map<String, JsonNode> getDefinition() {
        return definition;
    }
}`

File: src/test/java/uk/gov/hmcts/reform/iataskconfiguration/dmn/CamundaTaskInitiationTest.java (hunk 2 above)
Comment (indvamshi) [reply src/test/java/uk/gov/hmcts/reform/iataskconfiguration/dmn/CamundaTaskInitiationTest.java:501]: using same case as above.

File: src/main/resources/wa-task-initiation-ia-asylum.dmn (hunk 1 above)
Comment (alexbaeza) [reply src/main/resources/wa-task-initiation-ia-asylum.dmn:18]: let's take this offline

**Review by alexbaeza — APPROVED** [review]

**Pull request merged**